glossary.md

```diff
@@ -0,0 +1,37 @@
+
+**Merkle Tree**. A [tree](https://en.wikipedia.org/wiki/Tree_(data_structure)) data structure in which every leaf node is labelled with the [hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of a data block, and every non-leaf node is labelled with the [cryptographic hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function) of the labels of its child nodes.
+
+**Patricia Trie**. A special variant of the radix trie, in which rather than explicitly store every bit of every key, the nodes store only the position of the first bit which differentiates two sub-trees.
+
+**Merkle-Patricia Trie** Ethereum's cryptographically secure data structure, which combines characteristics of Merkle and Patricia tries to enable a reasonably efficient and verifiable representation of all accounts, balances, transactions, receipts, and state. See the [Ethereum wiki](https://eth.wiki/en/fundamentals/patricia-tree) for detailed information
+  * **Hexary trie**. Ethereum's current trie format is hexary: Each node in the trie has 16 children.
+  * **Binary trie**. Transitioning to a binary trie where each node has only two children is a major milestone in the Stateless Ethereum effort to bound witness sizes.
+
+**Polynomial commitment**
+
+**SNARK / STARK**
+
+**State** or **State Trie**. The Merkleized representation of Ethereum's shared execution state, which contains all account balances, smart contract memory and storage (storage is a nested trie within state and contains its own root). For more information see the [Ethereum wiki](https://eth.wiki/en/fundamentals/patricia-tree).
+
+
+**Statelessness** or **Statefullness**. The degree to which any Ethereum client participating in the Stateless Ethereum network keeps a local copy of state, or alternatively relies on a Witness for block execution.
+
+  * **Full-state** nodes keep a complete copy of state and of chain history, as all clients do currently.
+
+  * **Partial-state** or **semi-stateless**  keep a full state for just a short number of blocks, or perhaps just ‘watch’ the piece of state that they’re interested in, and get the rest of the data that they need to verify blocks from witnesses.
+
+  * **Zero-state** nodes keep no information about the state, and rely entirely on witnesses for block execution and tx validation.
+
+
+**Stateless Verification**
+
+**Witness**. The centerpiece of Stateless Ethereum: A data structure that 'proves' a valid state without containing the actual state data, only the missing hashes required to reproduce a particular `stateRoot`. See the [formal specification](./witness.md) for more information. Much of the work surrounding Stateless Ethereum is related to reducing the size of witnesses so that they can be useful on the network.
+
+**Witness compression**
```

Comment:
How about "Any method to reduce the size of a witness, which is considered a bottleneck to transaction throughput."?

Reply:
I guess I don't see why it's in the glossary if it's literally anything that reduces the size of a witness. I thought this had something to do with polynomial commitments and like IDK chunking up witnesses for a string of blocks so that un-touched state isn't wasting space.